### PR TITLE
Make small optimization in parser tests

### PR DESCRIPTION
**Pull request by iSazonov**
We call ExecuteCommand function over 80 times so it makes sense to optimize it for slow CI VMs.

### CODE REVIEW

File: test/powershell/Language/Parser/Parser.Tests.ps1 [hunk 1]
```diff
@@ -71,10 +71,10 @@ Describe "ParserTests (admin\monad\tests\monad\src\engine\core\ParserTests.cs)"
 		$functionDefinition>$functionDefinitionFile
 
         $PowerShell = [powershell]::Create()
+        $PowerShell.AddScript(". $functionDefinitionFile").Invoke()
```
Comment (vors) [inline test/powershell/Language/Parser/Parser.Tests.ps1]: there should be corresponding `$PowerShell.Commands.Clear()` to avoid re-executing it on the first run.
Also out of curiosity, did you measure the time before and after?

**Review by vors — CHANGES_REQUESTED** [review]

File: test/powershell/Language/Parser/Parser.Tests.ps1 (hunk 1 above)
Comment (iSazonov) [inline test/powershell/Language/Parser/Parser.Tests.ps1]: Fixed.
On my notebook:  25 ms * 84 = 2100 ms

Comment (vors) [inline test/powershell/Language/Parser/Parser.Tests.ps1]: is it before or after? how does it compare to the other?

**Review by vors — APPROVED** [review]

File: test/powershell/Language/Parser/Parser.Tests.ps1 (hunk 1 above)
Comment (iSazonov) [inline test/powershell/Language/Parser/Parser.Tests.ps1]: Sorry. Now I measure-command `ExecuteCommand` before and after patch - difference is ~8 ms on my notebook. So 8 ms * 84 = 676 ms. I don't measure memory consumption. 

Now I am surprised why two parser tests failed 😕

Comment (vors) [inline test/powershell/Language/Parser/Parser.Tests.ps1]: ah, I see you were talking about the diff.
It's useful to also see the relative numbers as well as absolute.

**Comment by iSazonov** [comment]
Pester report for Parser.Tests.ps1 on my notebook:
* before - 12.20s
* after    -  10.50s

Win is 1.7s

File: test/powershell/Language/Parser/Parser.Tests.ps1 [hunk 2]
```diff
@@ -518,6 +518,8 @@ Describe "ParserTests (admin\monad\tests\monad\src\engine\core\ParserTests.cs)"
 
 	It "Test that functions are resolved before cmdlets. (line 1678)"{
         $result = ExecuteCommand 'function testcmd-parserBVT { 3 };testcmd-parserBVT'
+        $PowerShell.AddScript(". $functionDefinitionFile").Invoke()
```
Comment (vors) [inline test/powershell/Language/Parser/Parser.Tests.ps1]: Why is this necessary?
It seems like a common setup that should be moved into "BeforeAll" or something.

**Review by vors — CHANGES_REQUESTED** [review]
Please, move the code into "BeforeAll"

File: test/powershell/Language/Parser/Parser.Tests.ps1 (hunk 2 above)
Comment (iSazonov) [inline test/powershell/Language/Parser/Parser.Tests.ps1]: It is a trick :-) Here ExecuteCommand redefine `testcmd-parserBVT` and we must to roll back the change otherwise subsequent tests will fail.

Comment (vors) [inline test/powershell/Language/Parser/Parser.Tests.ps1]: hm, that's not good. Can we just use something else, not `testcmd-parserBVT` in the above line?

Comment (iSazonov) [inline test/powershell/Language/Parser/Parser.Tests.ps1]: We could define a new cmdlet and the function directly in this test.

Comment (iSazonov) [inline test/powershell/Language/Parser/Parser.Tests.ps1]: We could define a new cmdlet and the function directly in this test.
Fixed.
